src/connector.ts

```diff
@@ -214,53 +226,63 @@ function resolveArguments(
 /**
  * See https://github.com/hasura/ndc-sdk-typescript for information on these interfaces.
  */
-export const connector: sdk.Connector<Configuration, Configuration, State> = {
+export const connector: sdk.Connector<RawConfiguration, Configuration, State> = {
   async try_init_state(
       config: Configuration,
       _metrics: unknown
   ): Promise<State> {
-    const functionsArg = resolve(config.functions || './functions/index.ts');
+    const functionsArg = config.inferenceConfig.functions;
     const functionsURL = `file://${functionsArg}`; // NOTE: This is required to run directly from deno.land.
     const functions = await import(functionsURL);
-    const info = getInfo(config);
     return {
-      functions,
-      info
+      functions
     }
   },
 
   get_capabilities(_: Configuration): sdk.CapabilitiesResponse {
     return CAPABILITIES_RESPONSE;
   },
 
-  make_empty_configuration(): Configuration {
-    const conf: Configuration = {
+  get_raw_configuration_schema(): JSONSchemaObject {
+    return RAW_CONFIGURATION_SCHEMA;
+  },
+
+  make_empty_configuration(): RawConfiguration {
+    const conf: RawConfiguration = {
       functions: './functions/index.ts',
       vendor: './vendor'
     };
     return conf;
   },
 
   // TODO: https://github.com/hasura/ndc-typescript-deno/issues/27 Make this add in the defaults
-  update_configuration(configuration: Configuration): Promise<Configuration> {
+  update_configuration(configuration: RawConfiguration): Promise<RawConfiguration> {
     return Promise.resolve(configuration);
   },
 
-  validate_raw_configuration(configuration: Configuration): Promise<Configuration> {
-    const defaults = {
-      preVendor: true,
+  validate_raw_configuration(configuration: RawConfiguration): Promise<Configuration> {
+    if (configuration.functions.trim() === "") {
```

Comment:
This seems like an odd check.

Reply:
You can't leave out functions any more. You must specify it.